shared/chat/conversation/list-area/normal/index.desktop.js

```diff
@@ -1,156 +1,162 @@
-          if (toFind === this.props.lastLoadMoreOrdinal) {
-            const idx = toFind ? this.props.messageOrdinals.indexOf(toFind) : -1
-            if (idx !== -1) {
-              const scrollToIdx = idx + 1
-              this._list.scrollToRow(scrollToIdx)
-            }
-          }
-        }
+    // Adjust scrolling
+    const list = this._listRef.current
+    // Prepending some messages?
+    if (snapshot && list && !this.state.isLockedToBottom) {
+      list.scrollTop = list.scrollHeight - snapshot
+    } else {
+      // maintain scroll to bottom?
+      if (this.state.isLockedToBottom && this.props.conversationIDKey === prevProps.conversationIDKey) {
+        this._scrollToBottom()
       }
     }
 
-    if (this.props.editingOrdinal && this.props.editingOrdinal !== prevProps.editingOrdinal) {
-      const idx = this.props.messageOrdinals.indexOf(this.props.editingOrdinal)
+    if (list && this.props.editingOrdinal && this.props.editingOrdinal !== prevProps.editingOrdinal) {
+      const ordinal = this.props.editingOrdinal
+      const idx = this.props.messageOrdinals.indexOf(ordinal)
       if (idx !== -1) {
-        this._list && this._list.scrollToRow(idx + 1)
+        const waypoints = list.querySelectorAll('[data-key]')
+        // find an id that should be our parent
+        const toFind = Types.ordinalToNumber(ordinal)
+        const found = Array.from(waypoints)
+          .reverse()
+          .find(w => parseInt(w.dataset.key, 10) < toFind)
+        if (found) {
+          found.scrollIntoView({behavior: 'smooth', block: 'center'})
+        }
       }
     }
   }
-  // we MUST debounce else this callbacks happens a bunch and we're switch back and forth as you submit
-  _updateBottomLock = debounce((clientHeight: number, scrollHeight: number, scrollTop: number) => {
-    // meaningless otherwise
-    if (clientHeight) {
-      this.setState(prevState => {
-        const isLockedToBottom = scrollTop + clientHeight >= scrollHeight - lockedToBottomSlop
-        return isLockedToBottom !== prevState.isLockedToBottom ? {isLockedToBottom} : null
-      })
-    }
-  }, 100)
-
-  _maybeLoadMoreMessages = debounce((clientHeight: number, scrollHeight: number, scrollTop: number) => {
-    if (clientHeight && scrollHeight && scrollTop <= 20) {
-      this.props.loadMoreMessages(this.props.messageOrdinals.first())
-    }
-  }, 500)
 
-  _onScroll = ({clientHeight, scrollHeight, scrollTop}) => {
-    this._updateBottomLock(clientHeight, scrollHeight, scrollTop)
-    this._maybeLoadMoreMessages(clientHeight, scrollHeight, scrollTop)
+  componentWillUnmount() {
+    this._cleanupDebounced()
   }
 
-  _onResize = ({width}) => {
-    if (this._cellCache.columnWidth({index: 0}) !== width) {
-      this._cellCache.clearAll()
+  _onResize = debounce(({bounds}) => {
+    if (this.state.isLockedToBottom) {
+      this._scrollToBottom()
     }
+
+    this.setState(p => (p.width === bounds.width ? null : {width: bounds.width}))
```

Comment:
I scrolled back 4 times or so and then made the screen a little wider, and the thread rerendered once but then was unresponsive for about a minute

Reply:
It's a little hard to repro, it might be that I'm catching this function multiple times so they end up rerendering on top of eachother. Maybe a longer debounce?

Reply:
yeah resizing is going to suck, unclear if there's a good fix actually

Reply:
lemme think on it, maybe i'll just remeasure the visible ones and just have more jank while you scroll